code/go/0chain.net/smartcontract/storagesc/blobber.go

```diff
@@ -71,209 +71,183 @@ func updateBlobberInList(list []*StorageNode, update *StorageNode) (ok bool) {
+	}
+
+	// set to zero explicitly, for "direct" calls
+	blobber.Capacity = 0
+
+	// remove from the all list, since the blobber can't accept new allocations
+	if savedBlobber.Capacity > 0 {
+		blobbers.Nodes.remove(blobber.ID)
+		sc.statIncr(statRemoveBlobber)
+		sc.statDecr(statNumberOfBlobbers)
 	}
 
-	return string(existingBlobber.Encode()), nil
+	return // opened offers are still opened
 }
 
-// addBlobber inserts, updates or removes a blobber; the blobber should
-// provide required tokens for stake pool; otherwise it will not be
+// inserts, updates or removes blobber
+// the blobber should provide required tokens for stake pool; otherwise it will not be
 // registered; if it provides token, but it's already registered and
 // related stake pool has required tokens, then no tokens will be
 // transfered; if it provides more tokens then required, then all
 // tokens left will be moved to unlocked part of related stake pool;
 // the part can be moved back to the blobber anytime or used to
 // increase blobber's capacity or write_price next time
 func (sc *StorageSmartContract) addBlobber(t *transaction.Transaction,
-	input []byte, balances cstate.StateContextI) (string, error) {
-
+	input []byte, balances cstate.StateContextI,
+) (string, error) {
 	// get smart contract configuration
 	conf, err := sc.getConfig(balances, true)
 	if err != nil {
-		return "", common.NewError("add_blobber_failed",
-			"can't get config: "+err.Error())
+		return "", common.NewError("add_or_update_blobber_failed",
+			"can't get config: " + err.Error())
 	}
 
-	// get registered "blobbers"
-	allBlobbers, err := sc.getBlobbersList(balances)
+	// get registered blobbers
+	blobbers, err := sc.getBlobbersList(balances)
 	if err != nil {
-		return "", common.NewError("add_blobber_failed",
-			"Failed to get blobber list: "+err.Error())
+		return "", common.NewError("add_or_update_blobber_failed",
+			"Failed to get blobber list: " + err.Error())
 	}
 
-	// create new "blobber"
+	// set blobber
 	var blobber = new(StorageNode)
 	if err = blobber.Decode(input); err != nil {
-		return "", common.NewError("add_blobber_failed",
-			"malformed request: "+err.Error())
+		return "", common.NewError("add_or_update_blobber_failed",
+			"malformed request: " + err.Error())
 	}
 
-	// edit new "blobber", setting transaction information
+	// set transaction information
 	blobber.ID = t.ClientID
 	blobber.PublicKey = t.PublicKey
 
-	if blobber.Capacity > 0 {
-		// insert or update "blobber"
-		err = sc.insertBlobber(t, conf, blobber, allBlobbers, balances)
-	} else {
-		// remove "blobber"
-		err = sc.removeBlobber(t, blobber, allBlobbers, balances)
-	}
-
-	if err != nil {
-		return "", common.NewError("add_blobber_failed", err.Error())
+	// insert, update or remove blobber
+	if err = sc.insertBlobber(t, conf, blobber, blobbers, balances); err != nil {
+		return "", common.NewError("add_or_update_blobber_failed", err.Error())
 	}
```

Comment:
This is an example of good place to write comment since its not obvious what the code is doing since "insertBlobber" also updates and removes blobber (lol). Even though this is an appropriate place to write comment its not a good comment since it focuses on "what", it would be better if it focused on "why".

Reply:
please, share code snippets written by you with "why" comments. to have a clue.